Jenkinsfile

```diff
@@ -1,90 +1,118 @@
                             }
                         }
                     }
+            }
+            parallel testJobs
+            boolean allPassed = true
+            int passed = 0
+            int failed = 0
+            echo "useMango Execution results: "
+            testResults.eachWithIndex { result, index ->
+                echo "${index + 1}. ${result.value}"
+                if (result.value.contains("Failed")){
+                    allPassed = false
+                    failed += 1
+                }
+                else {
+                    passed += 1
                 }
-                parallel testJobs
             }
-	    }
-	}
-}
-
-// Create a session on the server and return the authentication cookie
-def getAuthenticationCookie(String baseUrl, Object data) {
-    URL url = new URL("${baseUrl}/session")
-    HttpURLConnection conn = url.openConnection()
-    conn.setDoOutput(true)
-    conn.setRequestMethod("POST")
-	setRequestContent(conn, data)
-    conn.connect()
-    List<String> cookies = conn.getHeaderFields().get("Set-Cookie")
-    if (cookies.any()) {
-		return cookieValue(cookies.first())
-	}
-    else
-        throw new Exception("No cookies found")
-}
-
-def cookieValue(String cookieHeader) {
-	return cookieHeader.substring(0, cookieHeader.indexOf(";"))
+            echo "Total Executed: ${tests.size()}"
+            echo "Passed: ${passed}"
+            echo "Failed: ${failed}"
+            if (!allPassed){
+                error("Not all the tests passed.")
+            }
+        }
+    }
 }
 
-def setRequestContent(HttpURLConnection conn, Object data) {
-    conn.setRequestProperty("Content-Type", "application/json")
-    OutputStreamWriter out = new OutputStreamWriter(conn.getOutputStream())
-    out.write(JsonOutput.toJson(data))
-    out.flush()
-    out.close()
-}
 
-// Read all tests from the server whose folder matches the folder argument,
-// returning a list of the test names.
-def getTests(String baseUrl, String project, String folder, String authCookie) {
-    int pageSize = 20
-    int offset = 0
+// Read all tests with the tags specified
+def getTests(String baseUrl) {
+    String cursor = ""
     def tests = []
     def jsonSlurper = new JsonSlurper()
-    // loop to read tests page by page
+    echo "Retrieved the following tests from project ${params['Project']} with the tags ${params['Tags']} and status ${params['Status']}"
     while(true) {
-        URL url = new URL("${baseUrl}/projects/${project}/tests?offset=${offset}&pageSize=${pageSize}");
-        String content = url.getText(requestProperties:['Cookie':authCookie]);
+        URL url = new URL("${baseUrl}/projects/${params['Project']}/testindex?tags=${params['Tags']}&status=${params['Status']}&cursor=${cursor}")
+        HttpURLConnection conn = url.openConnection()
+        conn.setRequestMethod("GET")
+        conn.setDoInput(true)
+        conn.setRequestProperty("Authorization", "APIKEY ${params['Key']}")
```

Comment:
ApiKey should be stored as a secret credential, and not passed in as a parameter value.